core/local/steps/initial_diff.js

```diff
@@ -108,17 +109,21 @@ async function initialDiff (buffer /*: Buffer */, out /*: Buffer */, pouch /*: P
           byInode.delete(event.stats.fileid)
         }
         byInode.delete(event.stats.ino)
+        byPath.set(event.path, { path: event.path, kind: event.kind })
       } else if (event.action === 'initial-scan-done') {
         // Emit deleted events for all the remaining files/dirs
         for (const [, doc] of byInode) {
-          batch.push({
-            action: 'deleted',
-            kind: doc.kind,
-            _id: id(doc.path),
-            path: doc.path
-          })
+          if (!byPath.get(doc.path)) {
+            batch.push({
+              action: 'deleted',
+              kind: doc.kind,
+              _id: id(doc.path),
+              path: doc.path
+            })
```

Comment:
The code for this fix definitely looks good.
I need to find the corresponding scenario failing.